syntax = "proto3";

service WordCountService {
  rpc CountWords (TextRequest) returns (WordCountResponse);
  rpc MostFrequentWord (TextRequest) returns (MostFrequentWordResponse);
}

message TextRequest {
  string text = 1;
}

message WordCountResponse {
  map<string, int32> word_counts = 1;
}

message MostFrequentWordResponse {
  string word = 1;
  int32 count = 2;
}
